gc/base/PacketList.hpp

```diff
@@ -46,22 +46,40 @@ class MM_PacketList: public MM_BaseNonVirtual
 /* Data Section */
 public:
 	struct PacketSublist {
-		MM_Packet * _head;  /**< Head of the list */
-		MM_Packet * _tail;  /**< Tail of the list */
-		MM_LightweightNonReentrantLock _lock;  /**< Lock for getting/putting packets */
+		MM_Packet *_head;  /**< Head of the list */
+		MM_Packet *_tail;  /**< Tail of the list */
+		MM_LightweightNonReentrantLock *_lock;  /**< Lock for getting/putting packets */
 
 		bool
 		initialize(MM_EnvironmentBase *env)
 		{
-			if (!_lock.initialize(env, &env->getExtensions()->lnrlOptions, "MM_PacketList:_sublists[]._lock")) {
+			_lock = (MM_LightweightNonReentrantLock *)env->getExtensions()->getForge()->allocate(sizeof(MM_LightweightNonReentrantLock), OMR::GC::AllocationCategory::FIXED, OMR_GET_CALLSITE());
```

Comment:
I hope we can just copy the existing lock, rather than re-creating it.

Reply:
Reverted this, it simplified the code